lib/chef/cookbook_loader.rb

```diff
@@ -66,17 +66,27 @@ def merged_cookbook_paths # for deprecation warnings
       merged_cookbook_paths
     end
 
+    def warn_about_cookbook_shadowing
+      unless merged_cookbooks.empty?
+        Chef::Log.deprecation "The cookbook(s): #{merged_cookbooks.join(', ')} exist in multiple places in your cookbook_path. " +
+          "A composite version has been compiled.  In a future version of Chef this behavior will be removed."
```

Comment:
Probably?  Possibly?  I'm still debating in my head what to do about that.   This has to be Chef::Log and can't use the knife-ui stuff, but the other stuff is knife-ui...

I could just delete the knife-ui stuff but its nice and yellow and attention getting...